ospd/ospd.py

```diff
@@ -1188,11 +1188,21 @@ def handle_command(self, data: bytes, stream: Stream) -> None:
 
         response = command.handle_xml(tree)
 
+        write_success = True
         if isinstance(response, bytes):
-            stream.write(response)
+            write_success = stream.write(response)
         else:
             for data in response:
-                stream.write(data)
+                write_success = stream.write(data)
+                if not write_success:
+                    break
+
+        scan_id = tree.get('scan_id')
+        if self.scan_exists(scan_id) and command_name == "get_scans":
+            if write_success:
+                self.scan_collection.clean_temp_result_list(scan_id)
+            else:
+                self.scan_collection.restore_temp_result_list(scan_id)
```

Comment:
This looks a bit misplaced here to me but I don't have a better idea. This could need a lot of refactoring to get it into a better shape.

Reply:
Yes, I know. I am not happy with adding this code in this method neither. But we get the write problems here, not before.